backend/src/connector/mod.rs

```diff
@@ -1 +1,328 @@
+    ///
+    /// Returns `StatusCode::OK` on success
+    ///
+    /// # Errors
+    ///
+    /// The function fails if there was an error during creation of request
+    /// It shouldn't happen on normal behaviour
+    ///
+    pub async fn probe_main(&self) -> Result<StatusCode, ClientError> {
+        match self
+            .main
+            .get_nodes()
+            .await
+            .change_context(ClientError::Inaccessible)?
+        {
+            api::GetNodesResponse::AJSONArrayOfNodesInfoAndVdisksOnThem(_) => Ok(StatusCode::OK),
+            api::GetNodesResponse::PermissionDenied => Err(ClientError::PermissionDenied.into()),
+        }
+    }
+
+    /// Probes connection to the Bob's main connected node
+    ///
+    /// Returns `StatusCode::OK` on success
+    ///
+    /// # Errors
+    ///
+    /// The function fails if there was an error during creation of request
+    /// It shouldn't happen on normal behaviour
+    ///
+    pub async fn probe_secondary(&self, node_name: &NodeName) -> Result<StatusCode, ClientError> {
+        match self
+            .cluster
+            .get(node_name)
+            .ok_or(ClientError::NoClient)?
+            .get_nodes()
+            .await
+            .change_context(ClientError::Inaccessible)?
+        {
+            api::GetNodesResponse::AJSONArrayOfNodesInfoAndVdisksOnThem(_) => Ok(StatusCode::OK),
+            api::GetNodesResponse::PermissionDenied => Err(ClientError::PermissionDenied.into()),
+        }
+    }
+
+    /// Probes connection to the Bob's main connected node
+    ///
+    /// Returns `StatusCode::OK` on success
+    ///
+    /// # Errors
+    ///
+    /// The function fails if there was an error during creation of request
+    /// It shouldn't happen on normal behaviour
+    ///
+    // pub async fn probe_cluster(&self) -> Vec<(NodeName, StatusCode)> {
+    pub async fn probe_cluster(&self) -> Vec<(NodeName, StatusCode)> {
+        let v: Vec<_> = futures::stream::iter(&self.cluster)
+            .map(|(node_name, node)| async {
+                (
+                    node_name.clone(),
+                    match node.get_nodes().await {
+                        Ok(api::GetNodesResponse::AJSONArrayOfNodesInfoAndVdisksOnThem(_)) => {
+                            StatusCode::OK
+                        }
+                        Ok(api::GetNodesResponse::PermissionDenied) => StatusCode::UNAUTHORIZED,
+                        Err(_) => StatusCode::NOT_FOUND,
+                    },
+                )
+            })
+            .collect()
+            .await;
+
+        futures::future::join_all(v).await
+    }
+
+    /// Probes Node's connection
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if no client present for the specified Node's name,
+    /// the client was unable to receive response or the client doesn't have authority to do request.
+    pub async fn probe_socket(&self, name: &NodeName) -> Result<StatusCode, ClientError> {
```

Comment:
Duplicate? You have `probe_secondary` function that do exactly the same thing

Reply:
removed